File: VB/CustomSecurityOperation.Module/Controllers/RemoveBaseTypePermissionNewActionItemController.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy

Namespace CustomSecurityOperation.Module.Controllers
    Public Class RemoveBaseTypePermissionNewActionItemController
        Inherits ObjectViewController(Of ObjectView, PermissionPolicyTypePermissionObject)
        Protected Overrides Sub OnFrameAssigned()
            Dim _newObjectViewController As NewObjectViewController = Frame.GetController(Of NewObjectViewController)()
            If _newObjectViewController IsNot Nothing Then
                AddHandler _newObjectViewController.CollectDescendantTypes, Sub(s, e)
                                                                                e.Types.Remove(GetType(PermissionPolicyTypePermissionObject))
                                                                            End Sub

                AddHandler _newObjectViewController.ObjectCreating, Sub(s, e)
                                                                        If e.ObjectType = GetType(PermissionPolicyTypePermissionObject) Then
                                                                            e.NewObject = e.ObjectSpace.CreateObject(GetType(CustomTypePermissionObject))
                                                                        End If
                                                                    End Sub
            End If
            MyBase.OnFrameAssigned()
        End Sub
    End Class
End Namespace

File: CS/XPO/CustomPermission/CustomPermission.Module/Controllers/RemoveBaseTypePermissionNewActionItemController.cs
using CustomPermission.Module.Security;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.ExpressApp;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;

namespace CustomPermission.Module.Controllers {
    public class RemoveBaseTypePermissionNewActionItemController :
    ObjectViewController<ObjectView, PermissionPolicyTypePermissionObject> {
        protected override void OnFrameAssigned() {
            NewObjectViewController newObjectViewController = Frame.GetController<NewObjectViewController>();
            if (newObjectViewController != null) {
                newObjectViewController.CollectDescendantTypes += (s, e) => {
                    e.Types.Remove(typeof(PermissionPolicyTypePermissionObject));
                };
                newObjectViewController.ObjectCreating += (s, e) => {
                    if (e.ObjectType == typeof(PermissionPolicyTypePermissionObject)) {
                        e.NewObject = e.ObjectSpace.CreateObject(typeof(CustomTypePermissionObject));
                    }
                };
            }
            base.OnFrameAssigned();
        }
    }
}
